app/native/components/Alert.tsx

```diff
@@ -0,0 +1,24 @@
+import { grpcErrorStrings, ObscureGrpcErrorMessages } from "appConstants";
+import React from "react";
+import { ThemedText } from "@/components/ThemedText";
+
+interface AlertProps {
+  children: string;
+}
+
+export default function Alert({ children }: AlertProps) {
+  const oldErrorKey = Object.keys(grpcErrorStrings).find(
+    (oldError): oldError is ObscureGrpcErrorMessages =>
+      children.includes(oldError),
+  );
+
+  return (
+    <ThemedText type="error">
+      {
+        // Search for the error in the ugly grpc error object keys
+        // Replace it with the nice error if found
+        oldErrorKey ? grpcErrorStrings[oldErrorKey] : children
```

Comment:
What happens if the `oldErrorKey` is undefined? I think this will throw an error.